Test/CreateFile.Tests.ps1

```diff
@@ -0,0 +1,309 @@
+        Init
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: file.txt
+            Content: "First File."
+        
+        - CreateFile:
+            Path: file.txt
+            Content: "Second File... First File Overwritten"
+            Force: true
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenSuccess
+        ThenFile -Path 'file.txt' -Contains "Second File... First File Overwritten"
+    }
+}
+
+Describe -Name 'Createfile.when the path is outside root' {
+    It 'should not create the file' {
+        Init
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: ../file.txt
+            Content: "File above root."
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenTaskFailed
+        ThenError -Matches "'Path' given is outside of root. Please change one or more elements of the 'path'".
+    }
+}
+
+Describe -Name 'Createfile.when the path does not contain a file extension' {
+    It 'should create the file' {
+        Init
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: file
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenSuccess
+        ThenFile -Path 'file'
+    }
+}
+
+Describe -Name 'CreateFile.when a directory already exists with same path name' {
+    It -Name 'should not create the file'{
+        Init
+        GivenDirectory -Path 'file.txt'
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: file.txt
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenTaskFailed
+        ThenError -Matches "'Path' already points to a directory of the same name."
+    }
+ }
+
+ Describe -Name 'CreateFile.when a directory already exists with same path name and Force enabled' {
+    It -Name 'should not create the file'{
+        Init
+        GivenDirectory -Path 'file.txt'
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: file.txt
+            Force: True
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenTaskFailed
+        ThenError -Matches "'Path' already points to a directory of the same name."
+    }
+ }
+
+ Describe -Name 'CreateFile.when a subdirectory in the path does not exist' {
+    It -Name 'should not create the path' {
+        Init
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: \notreal\stillnotreal\file.txt
+            Content: 'I am not real.'
+'@
+        WhenRunningTask -ErrorAction SilentlyContinue
+        ThenTaskFailed
+        ThenError -Matches "'Path' contains subdirectories that do not exist."
+    }
+ }
+ 
+ Describe -Name 'CreateFile.when a subdirectory in path does not exist but Force is true' {
+    It -Name 'should create the entire path' {
+        Init
+        GivenWhiskeyYml @'
+        Build:
+        - CreateFile:
+            Path: \real\Iamreal\file.txt

```

Comment:
Add another `Describe` for the scenario where the given path exists but the parent item is a file not a directory, e.g. `Path: existingdirectory\IExistButImActuallyAFile.txt\file.txt`